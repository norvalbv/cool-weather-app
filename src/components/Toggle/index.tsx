import React, { ReactElement } from 'react';

export type ToggleProps = {
  labelLeft?: string;
  labelRight?: string;
  onclick?: () => void;
  isChecked: boolean;
};

const Toggle = ({ labelLeft, labelRight, onclick, isChecked }: ToggleProps): ReactElement => {
  const handleToggle = (): void => {
    if (onclick) onclick();
  };
  return (
    // eslint-disable-next-line jsx-a11y/label-has-associated-control
    <label className="mb-5 flex items-center gap-2">
      {labelLeft && (
        <span className={!isChecked ? 'text-violet-700 underline' : ''}>{labelLeft}</span>
      )}
      <div className="relative inline-flex cursor-pointer items-center">
        <input type="checkbox" className="peer sr-only" />
        <div
          onClick={handleToggle}
          className="bg-secondary peer h-5 w-9 rounded-full after:absolute after:left-[2px] after:top-[2px] after:h-4 after:w-4 after:rounded-full after:border after:border-gray-300 after:bg-white after:transition-all after:content-[''] peer-checked:after:translate-x-full peer-checked:after:border-white peer-focus:outline-none dark:border-gray-600 dark:bg-gray-700 dark:peer-focus:ring-blue-800"
        />
      </div>
      {labelRight && (
        <span className={isChecked ? 'text-violet-700 underline' : ''}>{labelRight}</span>
      )}
    </label>
  );
};

export default Toggle;
