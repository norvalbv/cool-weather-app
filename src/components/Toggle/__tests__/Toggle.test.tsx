import React, { FC } from 'react';
import renderer from 'react-test-renderer';

import Toggle, { ToggleProps } from '..';

const ToggleWithAllProps: FC<Required<ToggleProps>> = Toggle;

describe('<Toggle />', () => {
  test('Should render correctly with mandatory props', () => {
    const tree = renderer.create(<Toggle isChecked />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Should render correctly with all props', () => {
    const tree = renderer
      .create(
        <ToggleWithAllProps
          labelLeft="Test Label Left"
          labelRight="Test Label Right"
          onclick={jest.fn()}
          isChecked
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
