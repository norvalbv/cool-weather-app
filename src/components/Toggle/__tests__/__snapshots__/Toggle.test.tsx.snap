// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toggle /> Should render correctly with all props 1`] = `
<label
  className="mb-5 flex items-center gap-2"
>
  <span
    className=""
  >
    Test Label Left
  </span>
  <div
    className="relative inline-flex cursor-pointer items-center"
  >
    <input
      className="peer sr-only"
      type="checkbox"
    />
    <div
      className="bg-secondary peer h-5 w-9 rounded-full after:absolute after:left-[2px] after:top-[2px] after:h-4 after:w-4 after:rounded-full after:border after:border-gray-300 after:bg-white after:transition-all after:content-[''] peer-checked:after:translate-x-full peer-checked:after:border-white peer-focus:outline-none dark:border-gray-600 dark:bg-gray-700 dark:peer-focus:ring-blue-800"
      onClick={[Function]}
    />
  </div>
  <span
    className="text-violet-700 underline"
  >
    Test Label Right
  </span>
</label>
`;

exports[`<Toggle /> Should render correctly with mandatory props 1`] = `
<label
  className="mb-5 flex items-center gap-2"
>
  <div
    className="relative inline-flex cursor-pointer items-center"
  >
    <input
      className="peer sr-only"
      type="checkbox"
    />
    <div
      className="bg-secondary peer h-5 w-9 rounded-full after:absolute after:left-[2px] after:top-[2px] after:h-4 after:w-4 after:rounded-full after:border after:border-gray-300 after:bg-white after:transition-all after:content-[''] peer-checked:after:translate-x-full peer-checked:after:border-white peer-focus:outline-none dark:border-gray-600 dark:bg-gray-700 dark:peer-focus:ring-blue-800"
      onClick={[Function]}
    />
  </div>
</label>
`;
